### ISSUE TITLE
Port I3PhotonicsServiceCollection into trunk (Trac #2279)

### ISSUE THREAD

**Issue by icecube-trac** [icecube-trac/tix2#2271]
I3PhotonicsServiceCollection allows us to provide different splines for each new module type without having to modify existing reconstruction code. It should be brought into the trunk of photonics-service so we have it ready for the future.

Migrated from https://code.icecube.wisc.edu/ticket/2279
```json
{
    "status": "closed", 
    "changetime": "2019-07-09T15:36:26", 
    "description": "I3PhotonicsServiceCollection allows us to provide different splines for each new module type without having to modify existing reconstruction code. It should be brought into the trunk of photonics-service so we have it ready for the future.", 
    "reporter": "mjl5147", 
    "cc": "", 
    "resolution": "fixed", 
    "_ts": "1562686586421212", 
    "component": "upgrade", 
    "summary": "Port I3PhotonicsServiceCollection into trunk", 
    "priority": "normal", 
    "keywords": "", 
    "time": "2019-05-05T15:47:58", 
    "milestone": "", 
    "owner": "mjl5147", 
    "type": "enhancement"
}
```

**Comment by icecube-trac** [comment]
Trac update at 20190505T15:48:58: mjl5147 commented:

> Apparently I forgot this: it currently lives at https://code.icecube.wisc.edu/projects/icecube/browser/IceCube/projects/photonics-service/branches/mDOM-bulk-ice/public/photonics-service/I3PhotonicsServiceCollection.h

**Comment by icecube-trac** [comment]
Trac update at 20190709T15:36:26: 
* mjl5147 changed status from "new" to "closed"
* mjl5147 changed owner from "" to "mjl5147"
* mjl5147 commented:

> In changeset 174494:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="174494"
> Add I3PhotonicsCollection to the trunk of photonics-service so that we can use it with the upgrade/Gen2 reconstructions. Closes #2271
> ```


* mjl5147 changed resolution from "" to "fixed"
